python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -0,0 +1,217 @@
+# Wirepas Oy licensed under Apache License, Version 2.0
+#
+# See file LICENSE for full license details.
+#
+import queue
+import socket
+import ssl
+from select import select
+from threading import Thread, current_thread
+from time import sleep
+
+from paho.mqtt import client as mqtt
+
+
+class MQTTWrapper(Thread):
+    """
+    Class to manage the MQTT main thread and be able to share it with other services
+    In this case, it allows to have all the related mqtt activity happening on same thread
+    to avoid any dead lock from mqtt client.
+    """
+
+    def __init__(self, logger, username, password, host, port, secure_auth=True, tlsfile=None,
+                 on_termination_cb=None, on_connect_cb=None):
+        Thread.__init__(self)
+        self.daemon = True
+        self.running = False
+        self.logger = logger
+        self.on_termination_cb = on_termination_cb
+        self.on_connect_cb = on_connect_cb
+
+        self._client = mqtt.Client()
+        if secure_auth:
+            try:
+                self._client.tls_set(
+                    tlsfile,
+                    certfile=None,
+                    keyfile=None,
+                    cert_reqs=ssl.CERT_REQUIRED,
+                    tls_version=ssl.PROTOCOL_TLSv1_2,
+                    ciphers=None,
+                )
+            except:
+                self.logger.error(
+                    "Cannot use secure authentication. attempting unsecure connection"
+                )
+
+        self._client.username_pw_set(username, password)
+        self._client.on_connect = self._on_connect
+
+        try:
+            self._client.connect(host, port, keepalive=60)
```

Comment:
use a class variable to control the keep alive

Reply:
Done